File: Group6.java
import java.util.*;
public class Group6{

    public static void main(String[] args){
        Random random = new Random();
        Scanner scan = new Scanner(System.in);

        System.out.print("What is n? ");
        int n = scan.nextInt();
        System.out.print("Multiplied by? ");
        int x = scan.nextInt();

        int[][] array1 = new int[n][n];
        int[][] array2 = new int[n][n];

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                array1[i][j] = random.nextInt();

            }
        }

        int[] array3 = new int[n];

        for(int k=0; k<n; k++){
            array3[k] = random.nextInt();

        }

        int[][] array4 = new int[n][n];

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                array4[i][j] = random.nextInt();

            }
        }

        sum(n, array3);

        scalar(array1, array2, n, x);

        matrix(n, array1, array2, array4);

    }

    public static void matrix(int n, int[][] a1, int[][] a2, int[][] a3){
        int i,j,k;

        double timeStart = System.currentTimeMillis();

        for(i=0; i<n; i++)
            for(j=0; j<n; j++){
                a3[i][j] = 0;
                for(k=0; k<n; k++)
                    a3[i][j] = a3[i][j] + a1[i][k] * a2[k][j];
                
        }

        double timeEnd = System.currentTimeMillis();
        System.out.println("start: " + timeStart + "\nEnd " + timeEnd);
        double time = (timeEnd - timeStart);
        System.out.println("The time is: " + time + " ms" + "\nThe time is: " + (time/1000) + " sec");

    }

    public static void scalar(int[][] a1, int[][] a2, int n, int x){

        double timeStart = System.currentTimeMillis();

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                a2[i][j] = a1[i][j] * x;

            }
        }

        double timeEnd = System.currentTimeMillis();
        System.out.println("start: " + timeStart + "\nEnd " + timeEnd + "\narray2[2][3]: " + a2[2][3] + "\narray1[2][3]: " + a1[2][3]);
        double time = (timeEnd - timeStart);
        System.out.println("The time is: " + time + " ms" + "\nThe time is: " + (time/1000) + " sec");

    }
    
    public static int sum(int n, int[] a1){

        int p, result = 0;

        double timeStart = System.currentTimeMillis();

        for(p= 0; p< n; p++){

            result += a1[p];
        }

        double timeEnd = System.currentTimeMillis();
        System.out.println("start: " + timeStart + "\nEnd " + timeEnd);
        double time = (timeEnd - timeStart);
        System.out.println("The time is: " + time + " ms" + "\nThe time is: " + (time/1000) + " sec");

        return result;
    }



}

File: Group6.py
import random
from time import time, time_ns

n = input("Enter input size: ")
n = int(n)
x = input("Multiplied by? ")
x = int(x)
result = 0

list2d1 = [[random.random() for i in range(n)] for j in range(n)]
list2d2 = [[random.random() for i in range(n)] for j in range(n)]
list2d3 = [[random.random() for i in range(n)] for j in range(n)]
list2d4 = [[0 for i in range(n)] for j in range(n)]
list5 = [random.random() for i in range(n)]

#Matrix
timestartN = time_ns()
for i in range(n):
    for j in range(n):
        list2d3[i][j] = 0
        for k in range(n):
            list2d3[i][j] = list2d3[i][j] + (list2d1[i][k] * list2d2[k][j])


timeendN = time_ns()
print("start: " + str(timestartN) + "\nend: " + str(timeendN))
timedifN = timeendN - timestartN
print("The time is: " + str(timedifN/1000000) + " ms" + "\nThe time is: " + str(timedifN/1000000000) + " sec\n" + str(timedifN) + " nanosec")

#Scalar
timestartN = time_ns()
for i in range(n):
    for j in range(n):
        list2d4[i][j] = list2d1[i][j] * x


timeendN = time_ns()
print("start: " + str(timestartN) + "\nend: " + str(timeendN))
timedifN = timeendN - timestartN
print("The time is: " + str(timedifN/1000000) + " ms" + "\nThe time is: " + str(timedifN/1000000000) + " sec\n" + str(timedifN) + " nanosec")


#Sum
timestartN = time_ns()
for i in range(n):
    result += list5[i]


timeendN = time_ns()
print("start: " + str(timestartN) + "\nend: " + str(timeendN))
timedifN = timeendN - timestartN
print("The time is: " + str(timedifN/1000000) + " ms" + "\nThe time is: " + str(timedifN/1000000000) + " sec\n" + str(timedifN) + " nanosec")

